website/www/site/content/en/documentation/ml/preprocess-data.md

```diff
@@ -0,0 +1,227 @@
+
+### Artifact workflow {#artifact-workflow}
+
+The following scenario provides an example use case for artifacts.
+
+Before training a machine learning model, you set `artifact_mode` to `produce`.
+When you run `MLTransform`, it applies transformations that preprocess the
+dataset. The transformation produces artifacts that are stored in the location
+specified by the `artifact_location` parameter.
+
+After preprocessing, you use the transformed data to train the machine learning
+model.
+
+After training, you run inference. You use new test data and set
+`artifact_mode` to `consume`. By using this setting, you ensure that the test
+data undergoes the same preprocessing steps as the training data. In consume
+mode, running `MLTransform` fetches the transformation artifacts from the
+location specified in the `artifact_location` parameter. `MLTransform` applies
+these artifacts to the test data.
+
+This workflow provides consistency in preprocessing steps for both training and
+test data. This consistency ensures that the model can accurately evaluate the
+test data and maintain the integrity of the model's performance.
+
+## Preprocess data with MLTransform {#use-mltransform}
+
+To use the `MLTransform` transform to preprocess data, add the following code to
+your pipeline:
+
+```
+  import apache_beam as beam
+  from apache_beam.ml.transforms.base import MLTransform
+  from apache_beam.ml.transforms.tft import <TRANSFORM_NAME>
+  import tempfile
+
+  data = [
+      {
+          <DATA>
+      },
+  ]
+
+  artifact_location = tempfile.mkdtemp()
+  <TRANSFORM_FUNCTION_NAME> = <TRANSFORM_NAME>(columns=['x'])
+
+  with beam.Pipeline() as p:
+    transformed_data = (
+        p
+        | beam.Create(data)
+        | MLTransform(artifact_location=artifact_location).with_transform(
+            <TRANSFORM_FUNCTION_NAME>)
+        | beam.Map(print))
+```
+
+Replace the following values:
+
+-   TRANSFORM_NAME: The name of the [transform](#transforms) to use.
+-   DATA: The input data to transform.
+-   TRANSFORM_FUNCTION_NAME: The name that you assign to your transform
+    function in your code.
+
+For more examples, see
+[MLTransform for data processing](/documentation/transforms/python/elementwise/mltransform)
+in the [transform catalog](/documentation/transforms/python/overview/).
+
+### ScaleTo01 example {#scaleto01}
+
+This example demonstrates how to use `MLTransform` to normalize your data
+between 0 and 1 by using the minimum and maximum values from your entire
+dataset. `MLTransform` uses the `ScaleTo01` transformation.
+
+Use the following snippet to apply `ScaleTo01` on column `x` of the input
+data.
+
+```
+data_pcoll | MLTransform(artifact_location=<some_location>).with_transform(ScaleTo01(columns=['x']))
+```
+
+The `ScaleTo01` transformation produces two artifacts: the `min` and the `max`
+of the entire dataset. For more information, see the
+[Artifacts](#artifacts) section on this page.
+
+## Metrics {#metrics}
```

Comment:
I would like to add info explaining how to access the metrics.

Reply:
I will make a separate PR for this today or tomorrow.